### PR TITLE
MSR: Surface west cleanup

### PR DESCRIPTION
**Pull request by Miepee**

### CODE REVIEW

File: randovania/games/samus_returns/logic_database/Surface - West.txt
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Surface - West.txt:1]: Still has `Simple IBJ` to reach `Top of Mountain`. Needs IBJ Intermediate as discussed with a comment.

File: randovania/games/samus_returns/logic_database/Surface - West.txt [hunk 1]
```diff
@@ -78,22 +78,37 @@ Extra - asset_id: collision_camera_017
   * Layers: default
   * Extra - actor_name: LE_Platform_EnergyRecharge_001
   * Extra - actor_type: weightactivatedplatform
+  > Pickup (Missile Tank)
+      Any of the following:
+          # Spider Boost
+          Spider Boost (Beginner) and Can Spider Boost
+          # IBJ+Spider
+          Simple IBJ and Use Spider Ball
```
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Surface - West.txt:86]: Could also have `Space Jump and Mid-Air Morph (Beginner)` as a Spider path

File: randovania/games/samus_returns/logic_database/Surface - West.txt [hunk 2]
```diff
@@ -168,30 +186,29 @@ Extra - asset_id: collision_camera_017
...
+              Lightning Armor or Space Jump
               Baby Metroid or Morph Ball
   > Save Station
       All of the following:
           Baby Metroid
-          Wall Jump (Beginner) or Climb Rooms Vertically (High Jump)
+          Any of the following:
+              Single-wall Wall Jump (Beginner) or Wall Jump (Beginner) or Climb Rooms Vertically (High Jump)
+              Morph Ball and Super Jump (Intermediate) and Unmorph Extend (Intermediate)
+              # Damage Boost off the Gullug
```
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Surface - West.txt:198]: Gullugg

File: randovania/games/samus_returns/logic_database/Surface - West.txt [hunk 3]
```diff
@@ -168,30 +186,29 @@ Extra - asset_id: collision_camera_017
...
+              Morph Ball and Mid-Air Morph (Intermediate) and Damage Boost (Advanced) and Normal Damage ≥ 20
   > Pickup (Missile Tank)
```
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Surface - West.txt:200]: Should `Space Jump and Mid-Air Morph (Beginner) and Spider` also be added to this node or nah?

**Review by dyceron — CHANGES_REQUESTED** [review]

File: randovania/games/samus_returns/logic_database/Surface - West.txt (hunk 3 above)
Comment (Miepee) [reply randovania/games/samus_returns/logic_database/Surface - West.txt:200]: For `Below Ceiling Pickup` -> `Pickup (Missile Tank)`? I'd say no, if we already have the requirement in the other connection

**Review by dyceron — APPROVED** [review]

**Pull request merged**